internal/common/connectionsapi/client.go

```diff
@@ -0,0 +1,144 @@
+package connectionsapi
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/hashicorp/go-retryablehttp"
+)
+
+type Client struct {
+	authToken string
+	apiURL    url.URL
+	client    *http.Client
+}
+
+const (
+	defaultRetries = 3
+	defaultTimeout = 90 * time.Second
+)
+
+func NewClient(authToken string, rawAPIURL string, client *http.Client) (*Client, error) {
+	parsedAPIURL, err := url.Parse(rawAPIURL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse Metrics Endpoint API url: %w", err)
+	}
+
+	if client == nil {
+		retryClient := retryablehttp.NewClient()
+		retryClient.RetryMax = defaultRetries
+		client = retryClient.StandardClient()
+		client.Timeout = defaultTimeout
+	}
+
+	return &Client{
+		authToken: authToken,
+		apiURL:    *parsedAPIURL,
+		client:    client,
+	}, nil
+}
+
+type apiResponseWrapper[T any] struct {
+	Data T `json:"data"`
+}
+
+type MetricsEndpointScrapeJob struct {
+	Name                        string `json:"name"`
+	Enabled                     bool   `json:"enabled"`
+	AuthenticationMethod        string `json:"authentication_method"`
+	AuthenticationBearerToken   string `json:"bearer_token,omitempty"`
+	AuthenticationBasicUsername string `json:"basic_username,omitempty"`
+	AuthenticationBasicPassword string `json:"basic_password,omitempty"`
```

Comment:
Will these be expected to be encrypted by the HE API?

Reply:
@matthewnolf what do you think here? Do I need to encrypt these before sending them to Connections API?

Reply:
Done